src/css3.rs

```diff
@@ -1,120 +1,959 @@
+                        return Some(block);
+                    }
+
+                    self.token_stream.reconsume();
+                    if let Some(component_value) = self.consume_component_value() {
+                        block.values.push(component_value);
+                    }
+                }
+            }
+        }
+    }
+
+    fn consume_function(&mut self) -> Option<Function> {
+        let mut function = Function::new(self.token_stream.current().to_string());
+
+        loop {
+            match self.token_stream.consume() {
+                Token::RParen => {
+                    trace!("consume_function(): returning {:?}", function);
+                    break;
+                }
+                Token::Eof => {
+                    // parse error
+                    break;
+                }
+                _ => {
+                    self.token_stream.reconsume();
+                    if let Some(component_value) = self.consume_component_value() {
+                        function.values.push(component_value);
+                    }
+                }
+            }
+        }
+
+        Some(function)
+    }
+}
+
+
+
+pub struct CssAst {
+    tree: Node,
+}
+
+impl CssAst {
+    pub fn new() -> CssAst {
+        CssAst {
+            tree: Node {
+                name: "".to_string(),
+                attributes: HashMap::new(),
+                children: Vec::new(),
+            },
+        }
+    }
+
+    pub fn compile(&mut self, stylesheet: Stylesheet) {
+        let mut hm = HashMap::new();
+        hm.insert("location".to_string(), stylesheet.location.unwrap());
+
+        self.tree = Node {
+            name: "stylesheet".to_string(),
+            attributes: hm,
+            children: Vec::new(),
+        };
+
+        for rule in stylesheet.rules {
+            let node = match rule {
+                Rule::NormRule(norm_rule) => {
+                    self.compile_norm_rule(norm_rule)
+                }
+                Rule::AtRule(at_rule) => {
+                    self.compile_at_rule(at_rule)
+                }
+                Rule::QualifiedRule(qrule) => {
+                    self.compile_qualified_rule(qrule)
+                }
+            };
+
+            self.tree.children.push(node);
+        }
+    }
+
+    pub fn tree(&self) -> &Node {
+        &self.tree
+    }
+
+    pub fn display_tree(&self) {
+        self.display_tree_node(&self.tree, 0);
+    }
+
+    fn display_tree_node(&self, node: &Node, level: usize) {
+        println!("{}{}: {:?}", " ".repeat(level), node.name, node.attributes);
+        for child in &node.children {
+            self.display_tree_node(&child, level + 1);
+        }
+    }
+
+    fn compile_norm_rule(&mut self, _norm_rule: NormRule) -> Node {
+        let node = Node {
+            name: "norm-rule".to_string(),
+            attributes: HashMap::new(),
+            children: Vec::new(),
+        };
+
+        node
+
```

Comment:
```suggestion
        Node::new("norm-rule)
```
(same for the other functions under this one)